sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1113,6 +1111,17 @@ pi_result _pi_context::getAvailableCommandList(
     if (auto Res = Queue->insertActiveBarriers(CommandList, UseCopyEngine))
       return Res;
     return PI_SUCCESS;
+  } else {
+    // Cleanup regular command-lists if there are too many.
+    // It handles the case that the queue is not synced to the host
+    // for a long time and we want to reclaim the command-lists for
+    // use by other queues.
+    //
+    // TODO: rename ImmCmdListsEventCleanupThreshold to not mention
+    //       be for "immediate" only.
+    if (Queue->CommandListMap.size() > ImmCmdListsEventCleanupThreshold) {
```

Comment:
ImmCmdListsEventCleanupThreshold represents number of events in the command list but here you check number of command lists used in the queue. Do you think it worth having a separate threshold variable for that?